core/api/views/cp_records.py

```diff
@@ -290,6 +291,20 @@ def _get_new_cp_records(self, cp_report):
             "remarks": cp_report.comment,
         }
 
+        history = []
```

Comment:
This looks good, but I would move it to a method (or two) just like all other sections are implemented, to keep this method nice and slim.